include/nuttx/compiler.h

```diff
@@ -87,12 +87,29 @@
 #  define CONFIG_HAVE_CXX14 1
 #endif
 
+/* Green Hills Software definitions *****************************************/
+
+#if defined(__ghs__)
+
+#  define __extension__
+
+/* the following function is not truly need in runtime, and in order to fix
+ * undefined symbol link error, we need to make the following symbols as weak
+ */
+
+#  pragma weak __gh_long_long_printf
```

Comment:
Instead of adding ugly code to the architecture, you could refer to the tricore arch how to  handles unreferenced symbols , use a dummy c file to implement these symbols:
https://github.com/apache/nuttx/blob/master/arch/tricore/src/tc3xx/tc3xx_libc.c

Reply:
> Instead of adding ugly code to the architecture, you could refer to the tricore arch how to handles unreferenced symbols , use a dummy c file to implement these symbols: https://github.com/apache/nuttx/blob/master/arch/tricore/src/tc3xx/tc3xx_libc.c

I have carefully examined the directory structure of NuttX's code, but I did not find any directory specific to a particular compiler. 
Instead, I found code directories specific to certain architectures. 
If we want to create a special dummy implementation for GHS, can we create an implementation file like **"nuttx/compiler/ghs/ghs_dummy.c".**? is that work?

Reply:
Because this is a defect of the GHS compiler, we could place similar implementations in the following directory(same level as GNU): 
GNU:
https://github.com/apache/nuttx/tree/master/arch/arm/src/common/gnu
GHS (new):
https://github.com/apache/nuttx/tree/master/arch/arm/src/common/ghs